src/vma/sock/sockinfo_udp.cpp

```diff
@@ -926,7 +926,7 @@ int sockinfo_udp::setsockopt(int __level, int __optname, __const void *__optval,
 						break;
 					}
 
-					if ((NULL == __optval) || (__optlen > sizeof(struct ip_mreq_source) || (__optlen < sizeof(struct ip_mreq)))) {
+					if ((NULL == __optval) || (__optlen < sizeof(struct ip_mreq))) {
```

Comment:
@DanielLibenson later on (Line #934) you cast to `struct ip_mreq_source` so you must verify __optlen is big enough before using that part on memory.